packages/components/src/components/Select/Combobox.tsx

```diff
@@ -0,0 +1,166 @@
+import * as RadixPopover from '@radix-ui/react-popover';
+import { Slot as RadixSlot } from '@radix-ui/react-slot';
+import { useCombobox } from 'downshift';
+import { forwardRef, useRef, type ForwardedRef, type ReactNode } from 'react';
+
+import { SelectMenu } from './SelectMenu';
+import styles from './styles/select.module.scss';
+import { useSelectData, type SelectItemType } from './useSelectData';
+
+export type ComboboxProps = {
+    /**
+     * Children of the Combobox component. This can contain the `Select.Slot` components for the label, decorators, clear action and menu.
+     */
+    children?: ReactNode;
+    /**
+     * Callback function that is called when an item is selected.
+     */
+    onSelect?: (selectedItem: SelectItemType) => void;
+    /**
+     * The active item in the combobox component. This is used to control the combobox externally.
+     */
+    activeItem?: SelectItemType;
+    /**
+     * The default value of the combobox component. Used for uncontrolled usages.
+     */
+    defaultValue?: string;
+    /**
+     * The placeholder in the combobox component.
+     */
+    placeholder?: string;
+    /**
+     * Disables the combobox component.
+     */
+    disabled?: boolean;
+    /**
+     * The aria label of the combobox component.
+     */
+    'aria-label': string;
+    /**
+     * The data test id of the select component.
+     */
+    'data-test-id'?: string;
+};
+
+export const SelectCombobox = (
+    {
+        children,
+        onSelect,
+        activeItem,
+        defaultValue,
+        placeholder = '',
+        disabled,
+        'aria-label': ariaLabel,
+        'data-test-id': dataTestId = 'fondue-select-combobox',
+    }: ComboboxProps,
+    forwardedRef: ForwardedRef<HTMLDivElement>,
+) => {
+    const { inputSlots, menuSlots, items, defaultItem, filterText, clearButton, setFilterText } = useSelectData(
+        children,
+        defaultValue,
+    );
+
+    const {
+        getInputProps,
+        getToggleButtonProps,
+        getMenuProps,
+        getItemProps,
+        reset,
+        isOpen,
+        highlightedIndex,
+        inputValue,
+    } = useCombobox({
+        items,
+        onSelectedItemChange: ({ selectedItem }) => {
+            onSelect && onSelect(selectedItem);
+        },
+        selectedItem: activeItem,
+        onInputValueChange: ({ inputValue }) => {
+            setFilterText(inputValue);
+        },
+        defaultSelectedItem: defaultItem,
+        defaultHighlightedIndex: 0,
+        itemToString: (item) => (item ? item.label : ''),
+    });
+
+    const wasClicked = useRef(false);
+
+    return (
+        <RadixPopover.Root open={isOpen}>
+            <RadixPopover.Anchor asChild>
+                <div
+                    ref={forwardedRef}
+                    className={styles.root}
+                    data-error={
+                        inputValue && !items.find((item) => item.label.toLowerCase().includes(inputValue.toLowerCase()))
```

Comment:
memo the `items.find((item) => item.label.toLowerCase().includes(inputValue.toLowerCase()))` part?

Reply:
added